### PR TITLE
Allow to silently ignore signatures if the class is not found

### LINKED ISSUE #83: Add option to suppress warning about unresolved signature
I'm using the gradle plugin, version 2.0 . Please consider suppressing the warning about unresolved signature i.e. this signature:
`
Class 'com.google.inject.Inject' not found on classpath while parsing signature: com.google.inject.Inject [signature ignored]
`

The use case is as follows: 
- I have a multi-project gradle build, where some subprojects often use different libraries.
- I would like to have a single top-level forbidden signatures file.
- When a project starts using a library and also uses a forbidden method, i would like there to be an error. (e.g. in this case, I want to mandate that Guice users use the JSR-330 annotations instead of the guice annotations, however not all subprojects are using guice yet).

The `failOnUnresolvableSignatures = false` option makes the build still pass, but currently my build logs are polluted with messages about signatures for libraries from other projects. 

That being said, I just started using the plugin, and I'm open to ideas about how to better achieve my goals. Thank you for this useful plugin.

### PR DESCRIPTION
**Pull request by uschindler**
(e.g., multi-module builds with common signatures). The old setting failOnUnresolvableSignatures was deprecated in favour of ignoreSignaturesOfMissingClasses. This closes #83

### CODE REVIEW

File: src/main/java/de/thetaphi/forbiddenapis/Signatures.java [hunk 1]
```diff
@@ -170,6 +174,9 @@ private void addSignature(final String line, final String defaultMessage, final
       try {
         c = lookup.getClassFromClassLoader(clazz);
       } catch (ClassNotFoundException cnfe) {
+        if (this.ignoreSignaturesOfMissingClasses) {
```
Comment (uschindler) [inline src/main/java/de/thetaphi/forbiddenapis/Signatures.java:177]: This is the main point doing the magic!

**Comment by asdf2014** [comment]
Hi @uschindler . We found that after replacing the `failOnUnresolvableSignatures` old parameter with this `ignoreSignaturesOfMissingClasses` new parameter, it did not work well with Maven's Shade plugin. After adding the old parameters back, everything works fine. You can got more details from [here](https://github.com/apache/druid/pull/12158). PTAL, thanks a lot.

**Comment by uschindler** [comment]
Hi @asdf2014, 
Thanks for reporting. I have not yet looked into your code and why this does not work with renamed classes. I have no idea about the setup (so it looks like you shade one version of a library and then refer to the shaded version from source code?).

ignoreSignaturesOfMissingClasses only ignores missing classes, but failOnUnresolvableSignatures ignores any wrongly written signature (but it is deprecated now).

ignoreSignaturesOfMissingClasses works well when you have a subproject where a dependency is missing and you still want to use the same signatures file. In that case it just ignores all signatures where the class was not found. This should work also for shaded jars.

Do you have a simple example or maybe can you tell me how to trigger that after checking out Druid?

**Comment by uschindler** [comment]
Ah now I understand the problem:

[2022-01-15 05:49:23] [build-stdout] [2022-01-15 05:49:23] [autobuild] [ERROR] Failed to execute goal de.thetaphi:forbiddenapis:3.1:check (compile) on project opentelemetry-emitter: Parsing signatures failed: Method not found while parsing signature: com.google.common.util.concurrent.MoreExecutors#sameThreadExecutor() -> [Help 1]

So the class is there, but the method is missing (because it is another version of Google Guava). This of course fails. The idea behind the ignoreSignaturesOfMissingClasses is to just ignore signatures where class is missing. But this is not the case here.

The old setting was deprecated because it makes parsing signatures files a bit errorprone: If you have a typo in signatures file it won't fail the build.

**Pull request merged**